service/raft/raft_rpc.cc

```diff
@@ -222,7 +191,15 @@ future<raft::read_barrier_reply> raft_rpc::execute_read_barrier(raft::server_id
 }
 
 future<raft::snapshot_reply> raft_rpc::apply_snapshot(raft::server_id from, raft::install_snapshot snp) {
-    co_await _sm.transfer_snapshot(_address_map.get_inet_address(from), snp.snp);
+    auto ip_addr = _address_map.find(from);
+    if (!ip_addr.has_value()) {
+        // This is virtually impossible. We've just received the
+        // snapshot from the sender and must have updated our
+        // address map with its IP address.
+        const auto msg = format("Failed to apply snapshot from {}: ip address of the sender is not found", from);
+        co_return coroutine::exception(std::make_exception_ptr(raft::transport_error(msg)));
```

Comment:
If it's impossible, this should be `on_internal_error`.

Reply:
In theory there can be a race and a configuration change dropping the node right between request and response, so I'd handle it in a generic way.

Reply:
Okay, although even a configuration change should not give us problems, because it will only mark the entry as expiring, not completely remove it; there would have to be a 1-hour yield for us to hit this `if`.

On the other hand, if there was some serious problem with the address map causing entries to be missing when they shouldn't, `on_internal_error` would show up in tests (it would fail the test), which is desirable, exception - not necessarily.

Reply:
I think you're over thinking it. There is value in a general way of handling the issue of missing address. If the address map is buggy, the tests will show liveness issues. I can change it to your liking, but I think it's worthless.